### PR TITLE
Remove `api/`s that has never been supported

### PR DESCRIPTION
**Pull request by saschanaz**
#### Summary

Splitted from #6854 and covers changes only in `api/`. @foolip

#### Related issues

#6809

### CODE REVIEW

**Comment by saschanaz** [comment]
Hmm, probably better to split further?

File: api/PaymentAddress.json [hunk 1]
```diff
@@ -551,56 +551,6 @@
...
-            "edge": {
-              "version_added": false
-            },
-            "firefox": {
-              "version_added": false,
-              "notes": "Available only in nightly builds."
-            },
-            "firefox_android": {
-              "version_added": false,
-              "notes": "Available only in nightly builds."
```
Comment (saschanaz) [inline api/PaymentAddress.json:573]: There are several notes here like this, but I guess we can remove them since the interface itself is deprecated and thus we don't expected them to be shipped.

**Comment by foolip** [comment]
I'll review this as-is and if there are tricky bits those could be split further.

File: api/Document.json [hunk 2]
```diff
@@ -11217,54 +11217,6 @@
           }
         }
       },
-      "undoManager": {
```
Comment (foolip) [inline api/Document.json:11220]: This removal is correct, it's from https://github.com/WebKit/WebKit/blob/92ce6b62a589a0a70da09a6adfcfa03a07624467/Source/WebCore/dom/Document%2BUndoMananger.idl#L27 but hasn't been shipped.

File: api/Document.json [hunk 3]
```diff
@@ -11217,54 +11217,6 @@
...
-        "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Document/undoManager",
```
Comment (foolip) [inline api/Document.json:11222]: 404, nothing to clean up here 👍

File: api/Element.json [hunk 4]
```diff
@@ -8378,54 +8378,6 @@
           }
         }
       },
-      "tabStop": {
```
Comment (foolip) [inline api/Element.json:8381]: I can find no trace of this in Chromium, Gecko or WebKit so this removal seems correct.

File: api/Element.json [hunk 5]
```diff
@@ -8378,54 +8378,6 @@
...
-        "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Element/tabStop",
```
Comment (foolip) [inline api/Element.json:8383]: https://developer.mozilla.org/en-US/docs/Web/API/Element/tabStop will need to be removed too.

File: api/HTMLElement.json [hunk 6]
```diff
@@ -1079,54 +1079,6 @@
           }
         }
       },
-      "forceSpellCheck": {
-        "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLElement/forceSpellCheck",
```
Comment (foolip) [inline api/HTMLElement.json:1084]: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/forceSpellCheck will need to be removed.

File: api/HTMLElement.json [hunk 7]
```diff
@@ -1079,54 +1079,6 @@
           }
         }
       },
-      "forceSpellCheck": {
```
Comment (foolip) [inline api/HTMLElement.json:1082]: No trace of this in Chromium, Gecko or WebKit. In WebKit it appears in some performance tests, seemingly copied from the HTML spec where this once appeared:
https://github.com/whatwg/html/commit/4a0651c9d016dbf3a9431679248509baae6c20b3
https://github.com/whatwg/html/commit/c79ad811fb8b650c6079960642b86fe01effc3f7

File: api/PaymentAddress.json (hunk 1 above)
Comment (foolip) [reply api/PaymentAddress.json:573]: @Elchi3 do you have a link to the recent issue about representing preview browsers? I can't find it...

But in any case I agree with removing this one.

File: api/PaymentRequest.json [hunk 8]
```diff
@@ -509,58 +509,6 @@
           }
         }
       },
-      "paymentmethodchange_event": {
```
Comment (foolip) [inline api/PaymentRequest.json:512]: The `onpaymentmethodchange` entry shows support, maybe just copy that data?

File: api/ULongRange.json [hunk 9]
```diff
@@ -1,53 +0,0 @@
-{
-  "api": {
-    "ULongRange": {
```
Comment (foolip) [inline api/ULongRange.json:3]: This is like `ConstrainULong`, a long vs. unsigned long difference in spec and implementations. I'd break these two out into a separate PR.

File: api/XRPermissionDescriptor.json [hunk 10]
```diff
@@ -1,200 +0,0 @@
-{
-  "api": {
-    "XRPermissionDescriptor": {
```
Comment (foolip) [inline api/XRPermissionDescriptor.json:3]: This is a dictionary, which I don't think we should have in BCD but sometimes do. But whatever the rule on that, there's no trace of XRPermissionDescriptor in Chromium, Gecko or WebKit so this removal is fine.

File: api/XRPermissionStatus.json [hunk 11]
```diff
@@ -1,102 +0,0 @@
-{
-  "api": {
-    "XRPermissionStatus": {
```
Comment (foolip) [inline api/XRPermissionStatus.json:3]: I can't find this interface in Chromium, Gecko or WebKit except in tests/comments, so OK to remove.

File: api/XRPermissionStatus.json [hunk 12]
```diff
@@ -1,102 +0,0 @@
...
-      "__compat": {
-        "mdn_url": "https://developer.mozilla.org/docs/Web/API/XRPermissionStatus",
```
Comment (foolip) [inline api/XRPermissionStatus.json:5]: https://developer.mozilla.org/docs/Web/API/XRPermissionStatus will need to be removed.

File: api/XRPermissionDescriptor.json [hunk 13]
```diff
@@ -1,200 +0,0 @@
...
-      "__compat": {
-        "mdn_url": "https://developer.mozilla.org/docs/Web/API/XRPermissionDescriptor",
```
Comment (foolip) [inline api/XRPermissionDescriptor.json:5]: https://developer.mozilla.org/docs/Web/API/XRPermissionDescriptor will need to be removed.

**Review by foolip — COMMENTED** [review]
Thanks! Some of this could be merged now, and some needs to be paired with content updates. If you want to break this PR into smaller bits, I'd be happy to rubberstamp the result if it's just a subset of this.

File: api/PaymentAddress.json (hunk 1 above)
Comment (Elchi3) [reply api/PaymentAddress.json:573]: Removing sounds good to me, too. 
The proposal on "preview" is here: https://github.com/mdn/browser-compat-data/pull/10334 (and I think it would address this use case)

File: api/XRPermissionDescriptor.json (hunk 10 above)
Comment (saschanaz) [reply api/XRPermissionDescriptor.json:3]: >This is a dictionary, which I don't think we should have in BCD but sometimes do.

Is there a guideline for the "sometimes do" part?

**Comment by foolip** [comment]
@saschanaz will you also send a PR to remove the same bits on MDN, or would you like someone else to take care of it?

File: api/XRPermissionStatus.json (hunk 12 above)
Comment (saschanaz) [reply api/XRPermissionStatus.json:5]: It still exists in the spec and there is a tracking issue https://bugzilla.mozilla.org/show_bug.cgi?id=1582344, so maybe it should remain? 🤔

File: api/XRPermissionDescriptor.json (hunk 10 above)
Comment (foolip) [reply api/XRPermissionDescriptor.json:3]: My opinion is that they should never be in BCD, like mixins, but there's no guideline AFAIK. @ddbeck and @Elchi3 might have opinions too.

File: api/XRPermissionStatus.json (hunk 12 above)
Comment (foolip) [reply api/XRPermissionStatus.json:5]: Is it shipped anywhere yet?

Comment (saschanaz) [reply api/XRPermissionStatus.json:5]: Nah, but there have been requests to keep things that authors want to eventually ship, https://github.com/mdn/browser-compat-data/pull/6954#issuecomment-723939897 for example. #10619 is relevant here I guess.

File: api/XRPermissionDescriptor.json (hunk 10 above)
Comment (foolip) [reply api/XRPermissionDescriptor.json:3]: Splitting out this and the following XR thing would unblock this PR I think.

File: api/Element.json (hunk 5 above)
Comment (foolip) [reply api/Element.json:8239]: I'm creating a PR to remove this.

**Comment by foolip** [comment]
I've sent https://github.com/mdn/content/pull/5434 to go with this PR.

**Review by foolip — APPROVED** [review]
